docker/lib/dependabot/docker/file_parser.rb

```diff
@@ -181,6 +193,8 @@ def parse_helm(img_hash)
           []
         end
       end
+      # rubocop:enable Metrics/PerceivedComplexity
+      # rubocop:enable Metrics/AbcSize
```

Comment:
This got a bit out of hand, let me propose a simpler way:

```diff
-      # rubocop:disable Metrics/AbcSize
-      # rubocop:disable Metrics/PerceivedComplexity
       def parse_helm(img_hash)
         repo = img_hash.fetch("repository", nil)
         tag_value = img_hash.key?("tag") ? img_hash.fetch("tag", nil) : img_hash.fetch("version", nil)
@@ -178,22 +176,14 @@ module Dependabot
         tag = tag_details["tag"]
         digest = tag_details["digest"]
 
-        if !repo.nil? && !registry.nil? && !tag.nil? && !digest.nil?
-          ["#{registry}/#{repo}:#{tag}@sha256:#{digest}"]
-        elsif !repo.nil? && !registry.nil? && !tag.nil?
-          ["#{registry}/#{repo}:#{tag}"]
-        elsif !repo.nil? && !tag.nil? && !digest.nil?
-          ["#{repo}:#{tag}@sha256:#{digest}"]
-        elsif !repo.nil? && !tag.nil?
-          ["#{repo}:#{tag}"]
-        elsif !repo.nil?
-          [repo]
-        else
-          []
-        end
+        return [] unless repo
+        return [repo] unless tag
+
+        image = "#{repo}:#{tag}"
+        image.prepend("#{registry}/") if registry
+        image.append("@sha256:#{digest}/") if digest
+        [image]
       end
-      # rubocop:enable Metrics/PerceivedComplexity
-      # rubocop:enable Metrics/AbcSize
```

Also, I think we have no tests for parsing helm charts, not this PRs job but we should probably add some.

Reply:
Good call!  Will update and push changes.